SnapEngage/developer-documentation
# Sync from upstream lord/slate repo, and update Ruby gems to latest versions

# Description

Before I copy this repo's contents into our new private repo, we need to make sure we have all the latest from the original lord/slate repo.
----

Take a look at the README diff for explanations of the below actions.

1) Pulling down upstream changes from the lord/slate repo. This has not been done since we first forked the repo. There have only been a few bug fixes / feature improvements since then. (COMMIT: https://github.com/SnapEngage/developer-documentation/commit/fa02695e4099258e3009a9113a2086bcac2c27e7)
2) Updating our Gemfile with the latest versions of the gems (dependencies). (COMMIT:  https://github.com/SnapEngage/developer-documentation/commit/cc58aa6ae4dea35daa838f7aef4aed16ced307a0) After updating the Gemfile, I ran `make gems` which runs `bundle update`, which updates the gems, as well as writes out all updated versions to Gemfile.lock. Gemfile.lock is just a log of current dependency versions. More info here - http://bundler.io/v1.3/rationale.html
3) Updated some READMEs with the above info.
4) Updated Git pull request template with "How to test"

### Related Issue(s)
- n/a

### How to test

1) Open up the currently deployed version of the site: https://api-dev.snapengage.com/
2) Switch to this branch, (you'll be building it locally and smoke-check comparing it to the deployed version of the site to make sure nothing is horribly wrong.)
3) Follow the steps in the README to build and serve the site locally.
4) Do smoke-check 'ing. :)

Please run:

# Ready-to-Merge Checklist
- [x] Adds or updates dependencies?
- [x] Have you built these changes locally?
